### PR TITLE
Migration registration

### PR DESCRIPTION
**Pull request by Pytal**
Related: https://github.com/nextcloud/user_migration/pull/27

### To Do
- [x] Test
- [x] https://github.com/nextcloud/documentation/issues/8025

### CODE REVIEW

**Review by come-nc — APPROVED** [review]

File: lib/public/UserMigration/IExportDestination.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	/**
+	 * Adds a file to the export
+	 *
+	 * @param string $path Full path to the file in the export archive. Parent directories will be created if needed.
+	 * @param string $content The full content of the file.
+	 * @return bool whether the file was successfully added.
+	 *
+	 * @since 24.0.0
+	 */
+	public function addFile(string $path, string $content): bool;
```
Comment (come-nc) [inline lib/public/UserMigration/IExportDestination.php:45]: We should replace `addFile` by `addFileContents` and `addFileAsStream` to mirror `IImportSource`.
We did not need `addFileAsStream` in current codebase but might need it later for performance reasons or something, better safe than sorry.

**Review by PVince81 — APPROVED** [review]
:+1:

**Review by artonge — APPROVED** [review]

**Comment by Pytal** [comment]
Writing test

**Comment by Pytal** [comment]
Test added, would appreciate a re-review with the test :)

**Review by come-nc — APPROVED** [review]

**Pull request merged**